pulsar-broker/src/main/java/org/apache/pulsar/broker/offload/impl/BackedInputStreamImpl.java

```diff
@@ -76,26 +70,24 @@ private boolean refillBufferIfNeeded() throws IOException {
             long startRange = cursor;
             long endRange = Math.min(cursor + bufferSize - 1,
                                      objectLen - 1);
-            GetObjectRequest req = new GetObjectRequest(bucket, key)
-                .withRange(startRange, endRange);
-            log.debug("Reading range {}-{} from {}/{}", startRange, endRange, bucket, key);
-            try (S3Object obj = s3client.getObject(req)) {
-                versionCheck.check(key, obj.getObjectMetadata());
 
-                Long[] range = obj.getObjectMetadata().getContentRange();
-                long bytesRead = range[1] - range[0] + 1;
+            try {
+                Blob blob = blobStore.getBlob(bucket, key);
+                versionCheck.check(key, blob);
 
+                InputStream payload = blob.getPayload().openStream();
```

Comment:
try-with-resources around this payload.

When does the RPC happen with jclouds? Does openStream trigger reading anything from the network? The previous code only read the size of a block (1MB) at a time. This looks like it will read the whole object.

Reply:
Thanks, find a replacement, will change it.